### PR TITLE
Unit test error handlers

### PR DESCRIPTION
**Pull request by m4v15**
relates #71

### CODE REVIEW

File: tests/dbErrorHandlers.test.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+const dbErrorHandlers = require('../src/dbErrorHandlers.js')
+const { messages: errMessages, names: errNames, codes: errCodes } = require('../src/constants/errors.json')
+const { buildNextSpyTest, buildBoomSpyTest } = require('./helpers/index.js')
+
+const tape = require('tape')
+
+tape('dbErrorHandlers.custom with non custom error', t => {
+  buildNextSpyTest(t, dbErrorHandlers.custom, 'test error message', 'not a custom error')
```
Comment (des-des) [inline tests/dbErrorHandlers.test.js]: Okay, this is quite hard to read, or work out what the hell is going on, yes, it is good to keep tests dry, but also they should be very easy to read.

**Review by des-des — CHANGES_REQUESTED** [review]

File: tests/dbErrorHandlers.test.js (hunk 1 above)
Comment (des-des) [reply tests/dbErrorHandlers.test.js]: Okay,

I think:
```js
callErrorMiddleware = errorMiddleware => (error) => {
  const resSpy = 
  const nextSpy = 
  errorMiddleware(error, undefined, resSpy, nextSpy)
}
```

Then in your tests
```js
test('testing something', t => {
  const callMiddleware = callErrorMiddleware(middlewareToTest)

  const errorToTest = some error
  const { resSpy, nextSpy } = callMiddleware(errorToTest)

  // now, in the test function, we can make assertions in plain English
})
```

Okay that is very rough, but the idea is that you do not really need to look in the helpers to understand what assertions are being made. The helper should have names that makes it obvious what they are doing, and the assertions want to live in the test.

If you want to make helpers that make assertions, the most you want to be doing is having functions that are called in tests like
```js
assertSomethingRepetive(t, actual, expected, msg)
```

**Review by des-des — CHANGES_REQUESTED** [review]

**Comment by m4v15** [comment]
@des-des I've changed it now to be more like what you said. It's fairly repetitive, but I think in a good way to make the tests more readable.

Not sure on my function name, but it's sort of growing on me now...

File: tests/helpers/index.js [hunk 2]
```diff
@@ -1,7 +1,28 @@
...
+    }
+  }
+  return resSpy
+}
+
+helpers.spyGeneratorErrorMiddlewareCaller = errorMiddleware => (error) => {
+  const resSpy = buildResponseObj()
+  const nextSpy = sinon.spy()
+  errorMiddleware(error, undefined, resSpy, nextSpy)
+  return { resSpy, nextSpy }
```
Comment (des-des) [inline tests/helpers/index.js:27]: :+1: :

File: tests/helpers/index.js [hunk 3]
```diff
@@ -1,7 +1,28 @@
...
+    boom: {
+      badRequest: sinon.spy(),
+      notFound: sinon.spy(),
+      badImplementation: sinon.spy()
+    }
+  }
+  return resSpy
+}
+
+helpers.spyGeneratorErrorMiddlewareCaller = errorMiddleware => (error) => {
```
Comment (des-des) [inline tests/helpers/index.js:23]: I like the name

**Review by des-des — CHANGES_REQUESTED** [review]
Okay I have one change to request, it kind long though.

Make a folder called middlewar, then put each of the middlewares in its own file.

Finally mirror this in you test files.

Sorry not to have asked earlier.. Hope that okay!

look good though good work

**Review by des-des — APPROVED** [review]
